dali/base/dadfs.cpp

```diff
@@ -2738,6 +2738,20 @@ class CDistributedFileBase : implements INTERFACE, public CInterface
         return *t;
     }
 
+    IPropertyTree &queryHistory()
+    {
+        IPropertyTree *t = root->queryPropTree("History");
+        if (!t)
+            t = root->setPropTree("History",createPTree("History")); // takes ownership
```

Comment:
You should write with out a lock.

I suggest queryHistory() doesn't write and simply returns what's there, i.e. nullptr if never used.